File: ulohy/22-denni-rozvrh-ii/reseni/main.py
def read_int():
    return int(input())


def read_time():
    hours = read_int()
    minutes = read_int()
    return hours * 60 + minutes


if __name__ == "__main__":
    n = read_int()
    input();

    activity_names = []
    activity_start_times = []
    activity_end_times = []

    for _ in range(n):
        activity_names.append(input())
        activity_start_times.append(read_time())
        activity_end_times.append(read_time())
        input()

    current_time = read_time()

    activity_found = False
    for i in range(n):
        start_time = activity_start_times[i]
        end_time = activity_end_times[i]

        if (start_time <= current_time <= end_time) \
                or (start_time > end_time and (current_time >= start_time or current_time <= end_time)):
            activity_found = True
            print(activity_names[i])
            break

    if not activity_found:
        print("none")

File: ulohy/22-denni-rozvrh-ii/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let linesToReadCount;
let linesReadCount = 0;
const lines = [];

rl.on("line", (line) => {
    linesReadCount++;

    if (linesToReadCount === undefined) {
        linesToReadCount = parseInt(line) * 6 + 4;
    }

    lines.push(line);

    if (linesReadCount < linesToReadCount) {
        return;
    }

    rl.close();
    solve();
});

let nextLineIndex = 0;

function readLine() {
    nextLineIndex++;
    return lines[nextLineIndex - 1];
}

function readInt() {
    return parseInt(readLine());
}

function readTime() {
    const hours = readInt();
    const minutes = readInt();
    return hours * 60 + minutes;
}

function solve() {
    const n = readInt();
    readLine();

    const activityNames = [];
    const activityStartTimes = [];
    const activityEndTimes = [];

    for (let i = 0; i < n; i++) {
        activityNames.push(readLine());
        activityStartTimes.push(readTime());
        activityEndTimes.push(readTime());
        readLine();
    }

    const currentTime = readTime();

    let activityFound = false;
    for (let i = 0; i < n; i++) {
        const startTime = activityStartTimes[i];
        const endTime = activityEndTimes[i];

        if ((currentTime >= startTime && currentTime <= endTime)
            || (startTime > endTime && (currentTime >= startTime || currentTime <= endTime))) {
            activityFound = true;
            console.log(activityNames[i]);
            break;
        }
    }

    if (!activityFound) {
        console.log("none");
    }
}
